packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -254,10 +254,14 @@ export default util.createRule<Options, MessageId>({
           }
         }
 
-        if (isPromiseFinallyCallWithHandler(node)) {
-          return { isUnhandled: false };
+        // `x.finally()` is transparent to resolution of the promise, so check `x`.
+        // ("object" in this context is the `x` in `x.finally()`)
```

Comment:
[Praise] This second line was very helpful for understanding the code :) great comment!